lightning/src/chain/onchaintx.rs

```diff
@@ -646,14 +681,28 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 						// outpoints to know if transaction is the original claim or a bumped one issued
 						// by us.
 						let mut set_equality = true;
-						if request.outpoints().len() != tx.input.len() {
-							set_equality = false;
-						} else {
-							for (claim_inp, tx_inp) in request.outpoints().iter().zip(tx.input.iter()) {
-								if **claim_inp != tx_inp.previous_output {
-									set_equality = false;
+						if !request.requires_external_funding() {
+							if request.outpoints().len() != tx.input.len() {
+								set_equality = false;
+							} else {
+								for (claim_inp, tx_inp) in request.outpoints().iter().zip(tx.input.iter()) {
+									if **claim_inp != tx_inp.previous_output {
+										set_equality = false;
+									}
 								}
 							}
+						} else {
+							let request_inputs = request.outpoints();
```

Comment:
If we assume this logic is robust enough against consumer reordering transaction inputs, I think we can drop the no-ordering requirements as it can be a source of confusion for `HTLCResolution`. Rational of the lack of set-check for both variants could be commented.